### PR TITLE
Avoid mailing users that have explicitly unwatched an issue (#10475)

### PR DESCRIPTION
**Pull request by 6543**
backports #10475

### CODE REVIEW

File: models/issue_watch.go [hunk 1]
```diff
@@ -64,12 +64,16 @@ func getIssueWatch(e Engine, userID, issueID int64) (iw *IssueWatch, exists bool
...
+func GetIssueWatchersIDs(issueID int64, watching bool) ([]int64, error) {
+	return getIssueWatchersIDs(x, issueID, watching)
+}
+
+func getIssueWatchersIDs(e Engine, issueID int64, watching bool) ([]int64, error) {
 	ids := make([]int64, 0, 64)
-	return ids, x.Table("issue_watch").
+	return ids, e.Table("issue_watch").
 		Where("issue_id=?", issueID).
 		And("is_watching = ?", true).
```
Comment (guillep2k) [inline models/issue_watch.go]: shouldn't this be `And("is_watching = ?", watching).`?

Comment (6543) [reply models/issue_watch.go]: O you are right!

Comment (6543) [inline models/issue_watch.go]:
```suggestion
		And("is_watching = ?", watching).
```

**Comment by 6543** [comment]
@guillep2k done

**Review by guillep2k — APPROVED** [review]

**Comment by strk** [comment]
I understand this is a backport so LGTM (just to move things further) but it doesn't seem clean to call that getIssuesWatchers twice, rather I'd expect such a function to internally remove explicit unsubscribers...

**Comment by 6543** [comment]
@strk it doesnt work that way ... since you can "watch" via repo watch or participants or issuewatch ... and there are 3 types of issuewatch:
 * 1. No entry in table (so it use default witch is not subscribed unless you watch the repo or have participated)
* 2. IssueWatch.IsWatching = true; explicit watch
* 3. IssueWatch.IsWatching = false; explicit DONT watch

So i can query the userIDs in two functioncalls or query all IssueWatch entrys for a issue and itterate throu it fo set unset ...

**Pull request merged**